Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp

```diff
@@ -973,84 +976,86 @@
+          do_blocking_move_to_z(h_offset);                  // Move Z to the given 'H' offset before editing
         #endif
 
         KEEPALIVE_STATE(PAUSED_FOR_USER);
 
-        if (do_ubl_mesh_map) display_map(g29_map_type);             // Display the current point
+        if (do_ubl_mesh_map) display_map(g29_map_type);     // Display the current point
 
         ui.refresh();
 
-        float new_z = z_values[location.x_index][location.y_index];
-        if (isnan(new_z)) new_z = 0;                                // Invalid points begin at 0
-        new_z = FLOOR(new_z * 1000) * 0.001f;                       // Chop off digits after the 1000ths place
+        float new_z = z_values[lpos.x][lpos.y];
+        if (isnan(new_z)) new_z = 0;                        // Invalid points begin at 0
+        new_z = FLOOR(new_z * 1000) * 0.001f;               // Chop off digits after the 1000ths place
 
         lcd_mesh_edit_setup(new_z);
 
         do {
           new_z = lcd_mesh_edit();
           #if ENABLED(UBL_MESH_EDIT_MOVES_Z)
-            do_blocking_move_to_z(h_offset + new_z);                // Move the nozzle as the point is edited
+            do_blocking_move_to_z(h_offset + new_z);        // Move the nozzle as the point is edited
           #endif
           idle();
-          SERIAL_FLUSH();                                           // Prevent host M105 buffer overrun.
+          SERIAL_FLUSH();                                   // Prevent host M105 buffer overrun.
         } while (!ui.button_pressed());
 
-        if (!lcd_map_control) ui.return_to_status();                // Just editing a single point? Return to status
+        if (!lcd_map_control) ui.return_to_status();        // Just editing a single point? Return to status
 
-        if (click_and_hold(abort_fine_tune)) break;                 // Button held down? Abort editing
+        if (click_and_hold(abort_fine_tune)) break;         // Button held down? Abort editing
 
-        z_values[location.x_index][location.y_index] = new_z;       // Save the updated Z value
+        z_values[lpos.x][lpos.y] = new_z;                   // Save the updated Z value
         #if ENABLED(EXTENSIBLE_UI)
-          ExtUI::onMeshUpdate(location.x_index, location.y_index, new_z);
+          ExtUI::onMeshUpdate(location, new_z);
         #endif
 
-        serial_delay(20);                                           // No switch noise
+        serial_delay(20);                                   // No switch noise
         ui.refresh();
 
-      } while (location.x_index >= 0 && --g29_repetition_cnt > 0);
+      } while (lpos.x >= 0 && --g29_repetition_cnt > 0);
 
       ui.release();
 
       if (do_ubl_mesh_map) display_map(g29_map_type);
       restore_ubl_active_state_and_leave();
 
-      do_blocking_move_to(rx, ry, Z_CLEARANCE_BETWEEN_PROBES);
+      do_blocking_move_to(pos, Z_CLEARANCE_BETWEEN_PROBES);
```

Comment:
regression BUG: Z_CLEARANCE_BETWEEN_PROBES now used as feedrate instead of a Z-coordinate.